src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -556,9 +818,51 @@ class ov::genai::VLMPipeline::VLMPipelineImpl {
             
             ov::Tensor text_embeds = process_prompt(m_embedding, input_ids, m_vlm_config.scale_emb);
 
-            int64_t image_token_index = 32000; // TODO Consider getting from m_vlm_config.image_token_index or config.json
+            ov::Tensor encoded_image_token = m_tokenizer.encode(image_token, ov::genai::add_special_tokens(false)).input_ids;
+            int64_t image_token_id = encoded_image_token.data<int64_t>()[encoded_image_token.get_size() - 1];
+
+            return merge_text_and_image_embeddings_llava(input_ids, text_embeds, image_embeds, image_token_id);
+        }
+    }
+
+    ov::Tensor get_inputs_embeds_llava_next(const std::string& prompt, const std::vector<ov::Tensor>& images) {
+        std::string image_token = m_vlm_config.im_start;
+        // TODO Enable and reuse chat mode
+        std::string content = images.empty() ? prompt : image_token + "\n" + prompt;
+        ChatHistory m_history;
+        m_history.push_back({{"role", "user"}, {"content", content}});
+        constexpr bool add_generation_prompt = true;
+        std::string new_templated_chat_history;
+        try {
+            new_templated_chat_history = m_tokenizer.apply_chat_template(m_history, add_generation_prompt);
+        } catch (const std::exception& error) {
+            // TODO Consider using template syntax instead of concatenating
+            new_templated_chat_history = m_tokenizer.apply_chat_template(m_history, add_generation_prompt, "USER: " + content + " ASSISTANT:");
+        }
+         
+        ov::Tensor input_ids = m_tokenizer.encode(new_templated_chat_history).input_ids;
+        if (images.empty()) {
+            return process_prompt(m_embedding, input_ids, m_vlm_config.scale_emb);
+        } else {
+            OPENVINO_ASSERT(1 == images.size(), "Only a single image allowed");
```

Comment:
is it a current limitation of limitation of the model itself?

Reply:
This is current limitation

Reply:
Added to PR TODOs